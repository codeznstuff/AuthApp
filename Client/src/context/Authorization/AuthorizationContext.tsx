import React, { createContext, useContext } from 'react';
import { useAsync } from 'react-async';
import { fetchUser, IApplicationUser } from '../../services/AuthorizationApi';
import { getMsal } from '../../utils/authentication';
import { KiewitSpinner } from '../../components/common/KiewitSpinner/KiewitSpinner';
import UnauthenticatedApp from '../../UnauthenticatedApp';

export const AuthorizationContext = createContext<any>({
  currentUser: { UserId: 0 }
});

const loadData = async (): Promise<IApplicationUser | Error> => {
  try {
    // Make API call for MSAL AAD Authentication
    //const authUser = await getMsal();
    // Make API call to get user data from Authorization API
    //const user = await fetchUser(authUser.accountIdentifier);
    const user = await fetchUser('be64b27b-59b6-43bb-a824-51c16d328ffb');
    return user;
  } catch (error) {
    const ex: Error = error;
    return ex;
  }
};

export const AuthorizationProvider = (props: any): any => {
  const { data, isPending } = useAsync({
    promiseFn: loadData
  });

  if (isPending) {
    return <KiewitSpinner />;
  }
  if (data instanceof Error) {
    return <UnauthenticatedApp />;
  }
  return <AuthorizationContext.Provider value={data} {...props} />;
};

export const useAuthorization = (): IApplicationUser => {
  const context: any = useContext(AuthorizationContext);
  if (context === undefined) {
    throw new Error(`useAuthorization must be used within an AuthorizationProvider`);
  }
  return context;
};
