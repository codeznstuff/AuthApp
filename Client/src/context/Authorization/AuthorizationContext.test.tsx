import React from 'react';
import { render, RenderResult, cleanup } from '@testing-library/react';
import { AuthorizationContext } from './AuthorizationContext';

describe('AuthorizationContext', () => {
  let authorizationContext: RenderResult;
  const data: any = { user: { UserId: 0, Name: 'User0' } };

  beforeEach(() => {
    authorizationContext = render(<AuthorizationContext.Provider value={data} />);
  });

  afterEach(cleanup);

  it('matches the initial snapshot', () => {
    expect(authorizationContext.container.firstChild).toMatchSnapshot();
  });
});
