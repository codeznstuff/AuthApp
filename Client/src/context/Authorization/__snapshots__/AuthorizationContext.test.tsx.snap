// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthorizationContext matches the initial snapshot 1`] = `null`;
